src/plugins/liveobjects/liveobjectspool.ts

```diff
@@ -123,42 +126,84 @@ export class LiveObjectsPool {
     }
   }
 
+  applyBufferedStateMessages(bufferedStateMessages: BufferedStateMessage[]): void {
+    // since we receive state operation messages concurrently with the SYNC sequence,
+    // we must determine which operation messages should be applied to the now local copy of the object pool, and the rest will be skipped.
+    // since messages are delivered in regional order to the client, we can inspect the regional timeserial
+    // of each state operation message to know whether it has reached a point in the message stream
+    // that is no longer included in the state object snapshot we received from SYNC sequence.
+    for (const { regionalTimeserial, stateMessage } of bufferedStateMessages) {
+      if (!stateMessage.operation) {
+        this._client.Logger.logAction(
+          this._client.logger,
+          this._client.Logger.LOG_MAJOR,
+          'LiveObjects.LiveObjectsPool.applyBufferedStateMessages()',
+          `state operation message is received without 'operation' field, skipping message; message id: ${stateMessage.id}, channel: ${this._channel.name}`,
+        );
+        continue;
+      }
+
+      const existingObject = this.get(stateMessage.operation.objectId);
+      if (!existingObject) {
+        // for object ids we haven't seen yet we can apply operation immediately
+        this.applyStateMessages([stateMessage], regionalTimeserial);
+        continue;
+      }
+
+      // otherwise we need to compare regional timeserials
+      if (regionalTimeserial.before(existingObject.getRegionalTimeserial())) {
```

Comment:
This condition means we apply an operation if the operation timeserial >= the object timeserial.

The condition should instead be operation timeserial > the object timeserial, otherwise we risk double-counting an operation.

Reply:
Ah I see this is fixed in : https://github.com/ably/ably-js/pull/1912
Approved there

Reply:
yes, the comparsion is fixed both for origin and regional timeserials in the https://github.com/ably/ably-js/pull/1912